openmmtools/integrators.py

```diff
@@ -1054,104 +1063,14 @@ def __init__(self,
         # Compute constants
         kT = kB * temperature
         gamma = collision_rate
-        kinetic_energy = "0.5 * m * v * v"
-
-        # Convert splitting string into a list of all-caps strings
-        splitting = splitting.upper().split()
 
-        # Count how many times each step appears, so we know how big each R/V/O substep will be
-        n_R = sum([letter == "R" for letter in splitting])
-        n_V = sum([letter == "V" for letter in splitting])
-        n_O = sum([letter == "O" for letter in splitting])
-
-        # Check if the splitting string asks for multi-time-stepping.
-        # If so, each force group should be integrated for a total length equal to dt
-        if len(set([step for step in splitting if step[0] == "V"])) > 1:
-            mts = True
-            fgs = set([step[1:] for step in splitting if step[0] == "V"])
-            n_Vs = dict()
-            for fg in fgs:
-                n_Vs[fg] = sum([step[1:] == fg for step in splitting])
+        #check if integrator is metropolized by checking for M step:
+        if splitting.find("M") > -1:
+            self._metropolized_integrator = True
         else:
-            mts = False
+            self._metropolized_integrator = False
 
-        # Do a couple sanity checks on the splitting string
-        if override_splitting_checks == False:
-            # Make sure we contain at least one of R, V, O steps
-            assert ("R" in splitting)
-            assert ("V" in [s[0] for s in splitting])
-            assert ("O" in splitting)
-
-            # Make sure it contains no invalid steps
-            assert (set(splitting).issubset(set("RVO").union(set(["V{}".format(i) for i in range(32)]))))
```

Comment:
Hm, sorry--I interpreted "no invalid steps" to mean that it contained something other than RVO and the digits 0-9. Is that not correct?